### PR TITLE
LOG-1437: Added Continuous integration using Github actions

### PR DESCRIPTION
**Pull request by sangupta1**

### CODE REVIEW

File: .github/workflows/build-test-ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      with:
+        go-version: '^1.16.3' # The Go version to download (if necessary) and use.
+    
+    - name: Install dependencies
+      run: |
+        go mod download
+        
+    - name: Run Unit tests
+      run: |
+        go test -race -covermode atomic -coverprofile=covprofile ./pkg/...
```
Comment (igor-karpukhin) [inline .github/workflows/build-test-ci.yml]: @sangupta1 please use our Makefile target: `make test-unit`

Comment (sangupta1) [reply .github/workflows/build-test-ci.yml]: Sure @igor-karpukhin . I will make the change.

Comment (sangupta1) [reply .github/workflows/build-test-ci.yml]: @igor-karpukhin I have updated the PR. I have used the makefile target `make test` for the unit tests. There was not any target as `make test-unit` for this repository.

**Review by igor-karpukhin — APPROVED** [review]
Looks good

**Pull request merged**